### PR TITLE
[Enhancement] Iceberg metadata distributed plan e2e

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What I'm doing:

Fixes #issue
https://github.com/StarRocks/starrocks/issues/43460

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/runtime/metadata_result_writer.cpp [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    if (_sink_type == TResultSinkType::METADATA_ICEBERG) {
+        RETURN_IF_ERROR(_fill_iceberg_metadata(result_columns, chunk, result.get()));
+    }
+
+    return result;
+}
+
+// Columns is fixed in the first version of logical iceberg metadata table.
+// In principle, there will be no changes in the future.
+// Only new columns are allowed if necessary. The newly added columns need to check if nullable.
```
Comment (Youngwb) [inline be/src/runtime/metadata_result_writer.cpp:133]: If add new columns, at upgrage stage, BE version is higher than FE, would get not exist column here?

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 2]
```diff
@@ -476,9 +476,6 @@ public SerializedMetaSpec getSerializedMetaSpec(String dbName, String tableName,
 
         List<ManifestFile> deleteManifests = snapshot.deleteManifests(nativeTable.io());
         List<ManifestFile> matchingDeleteManifests = filterManifests(deleteManifests, nativeTable, predicate);
-        for (ManifestFile file : matchingDeleteManifests) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:479]: why remove these? how to knoew the matching delete manifest for BE

File: fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java [hunk 3]
```diff
@@ -379,8 +419,11 @@ public CloseableIterable<CombinedScanTask> planTasks() {
...
 
-    // For fe ut
-    public PlanMode getPlanMode() {
-        return planMode;
+    private int liveFilesCount(List<ManifestFile> manifests) {
+        return manifests.stream().mapToInt(this::liveFilesCount).sum();
+    }
+
+    private int liveFilesCount(ManifestFile manifest) {
+        return manifest.existingFilesCount() + manifest.addedFilesCount();
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java:427]: what's this meaning? why need to add?

File: fe/fe-core/src/main/java/org/apache/iceberg/MetadataParser.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+            fileScanTaskQueue.clear();
+        } finally {
+            while (true) {
+                if (fileScanTaskQueue.isEmpty()) {
+                    parserFinished.set(true);
+                    break;
+                }
+
+                try {
+                    Thread.sleep(50);
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/org/apache/iceberg/MetadataParser.java:136]: why need to sleep?

File: be/src/runtime/metadata_result_writer.cpp (hunk 1 above)
Comment (stephen-shelby) [reply be/src/runtime/metadata_result_writer.cpp:133]: The columns is come from `LogicalIcebergMetadataTable.toThrift`. So the jni scanner can't result the new column. we only need to check if nullable in here is enough.

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:479]: Currently we only support remote plan data file.  The delete file planning is local.

File: fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java (hunk 3 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java:427]: it's used for adapting iceberg metrics reporter. The metrics can be found by `trace values` or query profile.

File: fe/fe-core/src/main/java/org/apache/iceberg/MetadataParser.java (hunk 4 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/org/apache/iceberg/MetadataParser.java:136]: I think it's not a good idea that using `while(true)` to wait for the end. So I take a short sleep.

**Review by Youngwb — DISMISSED** [review]

**Review by Youngwb — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**